neo/Network/P2P/Payloads/WitnessScope.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Neo.Network.P2P.Payloads
+{
+    public enum WitnessScope : byte
+    {
+        /// <summary>
+        /// (neo2) - no params
+        /// </summary>
+        Global = 0x00,
+
+        /// <summary>
+        /// EntryOnly means that this condition must hold: EntryScriptHash == CallingScriptHash
+        /// No params is needed, as the witness/permission/signature given on first invocation will automatically expire if entering deeper internal invokes
+        /// This can be default safe choice for native NEO/GAS (previously used on Neo 2 as "attach" mode)
+        /// </summary>
+        EntryOnly = 0x01,
```

Comment:
Usually entry is `Transaction.Script`.

Reply:
Yes, logic is supposed to be: Hash160(Transaction.Script) == CallingScriptHash
Hash160(Transaction.Script) is EntryScriptHash, right?

Reply:
Ahhh, you mean the name? You prefer `EntryScriptHashOnly` ? But Entry is indeed referring to that Transaction.Script you mentioned... if you remembered it, perhaps its a good name :joy:

Reply:
`CalledByEntry`?

Reply:
`EntryOnly` or `CalledByEntry` ? Second is even more descriptive... first looks more "beautiful" to me :joy: but we can choose any for me.

Reply:
Nice name, agreed!